src/azure-cli/azure/cli/command_modules/synapse/commands.py

```diff
@@ -375,18 +379,22 @@ def get_custom_sdk(custom_module, client_factory):
         g.custom_command('cancel', 'cancel_spark_session_statement', confirmation=True)
 
     # Data Plane Commands --Access control operations
-    with self.command_group('synapse role assignment', synapse_accesscontrol_sdk,
+    with self.command_group('synapse role assignment', synapse_role_assignment_sdk,
                             custom_command_type=get_custom_sdk('accesscontrol', None)) as g:
         g.custom_command('create', 'create_role_assignment')
         g.custom_command('list', 'list_role_assignments')
         g.custom_show_command('show', 'get_role_assignment_by_id')
         g.custom_command('delete', 'delete_role_assignment', confirmation=True)
 
-    with self.command_group('synapse role definition', synapse_accesscontrol_sdk,
+    with self.command_group('synapse role definition', synapse_role_definitions_sdk,
                             custom_command_type=get_custom_sdk('accesscontrol', None)) as g:
         g.custom_command('list', 'list_role_definitions')
         g.custom_show_command('show', 'get_role_definition')
 
+    with self.command_group('synapse role scope', synapse_role_definitions_sdk,
+                            custom_command_type=get_custom_sdk('accesscontrol', None)) as g:
```

Comment:
You could declare your client factory here and don't need to define in every fuction.